src/v2/cookbook/form-validation.md

```diff
@@ -0,0 +1,322 @@
+
+``` js
+function main(args) {
+
+    return new Promise((resolve, reject) => {
+
+        // bad product names: vista,empire,mbp
+        let badNames = ['vista','empire','mbp'];
+        if(badNames.includes(args.name)) reject({error:'Existing product'});
+        resolve({status:'ok'});
+
+    });
+
+}
+```
+
+Basically any name but "vista", "empire", and "mbp" are acceptable. Ok, so let's look at the form.
+
+``` html
+<form id="app" @submit="checkForm" method="post">
+  
+  <p v-if="errors.length">
+    <b>Please correct the following error(s):</b>
+    <ul>
+      <li v-for="error in errors">{{ error }}</li>
+    </ul>
+  </p>
+
+  <p>
+  <label for="name">New Product Name: </label>
+  <input type="text" name="name" id="name" v-model="name">
+  </p>
+
+  <p>
+  <input type="submit" value="Submit">  
+  </p>
+
+</form>
+```
+
+There isn't anything special here. So let's go on to the JavaScript.
+
+``` js
+const apiUrl = 'https://openwhisk.ng.bluemix.net/api/v1/web/rcamden%40us.ibm.com_My%20Space/safeToDelete/productName.json?name=';
+
+const app = new Vue({
+  el:'#app',
+  data:{
+    errors:[],
+    name:''
+  },
+  methods:{
+    checkForm:function(e) {
+      e.preventDefault();
+      this.errors = [];
+      if(this.name === '') {
+        this.errors.push("Product name is required.");
+      } else {
+        fetch(apiUrl+encodeURIComponent(this.name))
+        .then(res => res.json())
+        .then(res => {
+          if(res.error) {
+            this.errors.push(res.error);
+          } else {
+            // redirect to a new url, or do something on success
+            alert('ok!');
+          }
+        });
+      }
+    }
+  }
+})
+```
+
+We start off with a variable representing the URL of the API that is running on OpenWhisk. Now look at `checkForm`. In this version, we always prevent the form from submitting (which, by the way, could be done in the HTML with Vue as well). You can see a basic check on this.name being empty, and then we hit the API. If it's bad, we add an error as before. If it's good, right now we do nothing (just an alert), but you could navigate the user to a new page with the product name in the URL, or do other actions as well. You can run this demo below:
+
+<p data-height="265" data-theme-id="0" data-slug-hash="BmgzeM" data-default-tab="js,result" data-user="cfjedimaster" data-embed-version="2" data-pen-title="form validation 4" class="codepen">See the Pen <a href="https://codepen.io/cfjedimaster/pen/BmgzeM/">form validation 4</a> by Raymond Camden (<a href="https://codepen.io/cfjedimaster">@cfjedimaster</a>) on <a href="https://codepen.io">CodePen</a>.</p>
+<script async src="https://production-assets.codepen.io/assets/embed/ei.js"></script>
+
+## Vue Form Validation Libraries

```

Comment:
I'd appreciate a little more consideration here. Thanks for including it at our request, but we'd call this section "Alternative Patterns" and I'd love it if it addressed _why_ you might use a library in place of what you've written above and what the tradeoffs are (one is larger, but it might cover more use cases for different types of forms if you know you're building out more forms later. They will probably stay up to date with browser support based on community improvements etc)

Thanks!